server/login.cpp

```diff
@@ -1135,6 +1183,13 @@ bool Create::getDeity(const std::shared_ptr<Socket>& sock, std::string str, int
 
         for(l=1, k=0; l < static_cast<int>(CreatureClass::CLASS_COUNT)+4; l++) {
 
+            //Gender restrict for Mara, Arachnus, Linothan clerics
+                if( sock->getPlayer()->getClass() == CreatureClass::CLERIC &&
+                    ((sock->getPlayer()->flagIsSet(P_MALE) && l == MARA) ||
+                        (sock->getPlayer()->flagIsSet(P_MALE) && l == ARACHNUS) ||
+                            (!sock->getPlayer()->flagIsSet(P_MALE) && l == LINOTHAN)))
```

Comment:
Same comment as above

Reply:
It is done to exclude the wrong option. If male it wont give the Mara option to choose. If female, it wont give the Linothan option to choose. When Arachnus goes in, if you're male, it won't show the Arachnus option to choose. 

This is repeated here because it's in the section that does the work of processing the choice. Without it, you'll have an invisible option and can still choose even when you can't see it, plus it will break and makes you always choose Linothan even if you chose Mara.